src/oci.c

```diff
@@ -523,6 +526,90 @@ cc_oci_attach (struct cc_oci_config *config,
 }
 
 /*!
+ * Get the home directory for the workload user
+ *
+ * \param config \ref cc_oci_config.
+ *
+ * \return home dir for user on sucess, "/" by default
+ */
+
+private gchar*
+get_user_home_dir(struct cc_oci_config *config) {
+	gchar          *user_home = NULL;
+	gchar          *password_path = NULL;
+	FILE           *pw_file = NULL;
+	struct passwd  *pw_entry;
+
+	if (! config) {
+		goto end;
+	}
+
+	password_path = g_strdup_printf ("%s/%s", config->oci.root.path, "etc/passwd");
+	pw_file = g_fopen (password_path, "r");
+	if ( pw_file == NULL) {
+		goto end;
+	}
+
+	while ((pw_entry = fgetpwent(pw_file)) != NULL) {
+		if (pw_entry->pw_uid == config->oci.process.user.uid) {
+			user_home = strdup(pw_entry->pw_dir);
+			break;
+		}
+	}
+
+end:
+	if (! user_home) {
+		user_home = strdup("/");
+	}
+	g_free_if_set(password_path);
+
+	if (pw_file) {
+		fclose(pw_file);
+	}
+	return user_home;
+}
+
+/*!
+ * Set the HOME environment variable
+ *
+ * \param config \ref cc_oci_config.
+ *
+ * returns early if HOME is present in the env config
+ */
+private void
+set_env_home(struct cc_oci_config *config)
+{
+	gchar *user_home_dir = NULL;
+
+	if (! (config && config->oci.process.env)) {
+		return;
+	}
+
+	/* Check if HOME is set in the environment config */
+	gchar **var = config->oci.process.env;
+	while (*var != NULL) {
```

Comment:
This might be better as a `for (gchar **var = config->oci.process.env; ...)` loop - `var` would then also be further localised.